interface_service/src/pubsub.py

```diff
@@ -0,0 +1,77 @@
+"""Provides a `subscribe` function
+   That creats a new subscription to a Pub/Sub topic.
+"""
+
+import uuid
+
+from google.cloud import pubsub_v1
+from google.api_core.exceptions import AlreadyExists
+
+from src.config import settings
+from src.var_names import VarNames
+
+
+def subscribe(unique_subscription_name: bool = False):
+    """Subscribes to a Pub/Sub topic.
+
+    Args:
+        unique_subscription_name (bool, optional): Whether to create a unique subscription name.
+        This should be enabled for the interface services. Defaults to False.
+    """
+
+    def callback(message: pubsub_v1.subscriber.message.Message) -> None:
+        """Acknowledges a Pub/Sub message. Used in the `subscribe()` function.
+
+        Args:
+            message (pubsub_v1.subscriber.message.Message): The message
+            to acknowledge.
+        """
+        message.ack()
+
+    # Create the client
+    subscriber = pubsub_v1.SubscriberClient()
+
+    # Wrap the subscriber in a 'with' block to automatically call close() to
+    # close the underlying gRPC channel when done.
+    with subscriber:
+
+        # Get the topic path
+        topic_path = subscriber.topic_path(
```

Comment:
There should be some code to check if the topic exists - it will also do on the actual environment, since it's created by Terraform, but the emulator *could* use it.